util/image_viewer/spotfinder_frame.py

```diff
@@ -1470,9 +1521,10 @@ def map_coords(x, y, p):
                     if self.settings.show_ctr_mass and "xyzobs.px.value" in reflection:
                         centroid = reflection["xyzobs.px.value"]
                         # ticket #107
-                        if centroid[2] >= i_frame and centroid[2] <= (
-                            i_frame + self.params.stack_images
-                        ):
+                        if (
+                            centroid[2] >= i_frame
+                            and centroid[2] <= (i_frame + self.params.stack_images)
+                        ) or self.viewing_stills:
```

Comment:
short-circuit evaluation, plus makes reading easier if you're in `viewing_stills` mode.
```suggestion
                        if self.viewing_stills or (
                            i_frame <= centroid[2] <= (i_frame + self.params.stack_images)
                        ):
```